src/main/java/com/deavensoft/paketomat/center/CenterServiceImpl.java

```diff
@@ -54,6 +62,49 @@ public void updateStatus(Long code, Status status) {
             }
         }
     }
+    public void payment(Long id, Paid paid)
+    {
+        List<Package> packages = centerRepository.findAll();
+        for (Package p : packages) {
+            if(p.getId().equals(id)){
+                p.setPaid(paid);
+                sendMailToUser(p.getUser().getEmail(),paid);
+                centerRepository.save(p);
+            }
+        }
+
+
+    }
+
+    public void sendMailToUser(String email, Paid p) {
+        EmailDetails emailSender = new EmailDetails();
+        emailSender.setRecipient(email);
+        if (Paid.PAID == p) {
+            emailSender.setMsgBody("Your package is in the paketomat and is ready to be picked up and the code is" +" "+
+                    generateCode());
+        } else if (Paid.NOT_PAID == p) {
+            emailSender.setMsgBody("Your package is in the paketomat and is ready to be paid");
+        } else if (Paid.UN_SUCESSFULL == p) {
+            emailSender.setMsgBody("Your package is in the paketomat, the payment was unsuccesfull, try again to pay for the package");
+        }
+        emailSender.setAttachment("");
+        emailSender.setSubject("Package arrived in the paketomat");
+        Map<String, Object> model = new HashMap<>();
+        model.put("msgBody", emailSender.getMsgBody());
+        emailService.sendMailWithTemplate(emailSender, model);
+        log.info("E-Mail is sent to the end user");
+
+    }
```

Comment:
You don't need this implementation. Just call the appropriate method from the emailService.